File: Diagrammatize.Design.Pattern/code/cxx/Bridge/A2/RandomCountDisplay.cpp
#include "RandomCountDisplay.hpp"
#include <chrono>

RandomCountDisplay::RandomCountDisplay(DisplayImpl* impl): 
    CountDisplay(impl), 
    random(std::chrono::system_clock::now().time_since_epoch().count()) {
}

void RandomCountDisplay::randomDisplay(int times) {
    multiDisplay(random.randomInt(0, times-1));
}

File: Diagrammatize.Design.Pattern/code/python/Bridge/A1/RandomCountDisplay.py
import time
import random

from CountDisplay import CountDisplay

class RandomCountDisplay(CountDisplay):
    def __init__(self, impl):
        super().__init__(impl)
        self.random = random.Random(time.time())

    def randomDisplay(self, times):
        self.multiDisplay(self.random.randint(0, times-1))
